go/vm/evmc/evmc_interpreter.go

```diff
@@ -55,6 +58,10 @@ func (e *EvmcInterpreter) Run(params vm.Parameters) (vm.Result, error) {
 		context: params.Context,
 	}
 
+	if params.Revision > NewestSupportedRevision {
+		return vm.Result{}, &vm.ErrUnsupportedRevision{}
```

Comment:
You set up the `ErrUnsupportedRevision` error as a type to allow you to pass along the unsupported revision.

If this feature of the `ErrUnsupportedRevision` error is not used, it might be simpler to define a error constant like [here](https://github.com/Fantom-foundation/Tosca/blob/8b6197ed9a575a1944dcbe524c74a9b42cb59e34/go/ct/gen/error.go#L19). This makes the returning of the error simpler (not & and {}) and checking for it is easier using `errors.Is`.